src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,308 @@
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_softmaxcrossentropyloss : op_parser<parse_softmaxcrossentropyloss>
+{
+    std::vector<op_desc> operators() const { return {{"SoftmaxCrossEntropyLoss"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd */,
+                                       const onnx_parser& parser,
+                                       const onnx_parser::node_info& info,
+                                       const std::vector<instruction_ref>& args) const
+    {
+        // Get and handle attributes
+        std::string reduction = "mean";
+        if(contains(info.attributes, "reduction"))
+        {
+            reduction = info.attributes.at("reduction").s();
+            if(not contains({"mean", "sum", "none"}, reduction))
+            {
+                MIGRAPHX_THROW("Invalid reduction mode: " + reduction +
+                               "\n Valid options are [none, mean, sum]");
+            }
+        }
+
+        // ignore_index is optional attribute, assign this as a scalar literal input to the op
+        auto ignore_index =
+            info.add_literal(migraphx::literal(migraphx::shape(shape::int64_type, {1}, {0}), {-1}));
+        bool has_ignore_index = contains(info.attributes, "ignore_index");
+        if(has_ignore_index)
+        {
+            auto ignore_index_val =
+                parser.parse_value(info.attributes.at("ignore_index")).at<int64_t>();
+            ignore_index = info.add_literal(migraphx::literal(
+                migraphx::shape(migraphx::shape::int64_type, {1}, {0}), {ignore_index_val}));
+        }
+
+        // Get and validate Inputs
+        auto scores = args.at(0);
+        auto labels = args.at(1);
+        auto scores_shape = scores->get_shape();
+        auto label_shape = labels->get_shape();
+
+        if(scores_shape.ndim() < 2)
+        {
+            MIGRAPHX_THROW("softmaxcrossentropyloss: Scores must be two or more dimensions [batch, "
+                           "class_size, D1...Dk]");
+        }
+
+        // TODO: Update this operator when we get bfloat16 support
+        std::set<migraphx::shape::type_t> supported_score_weight_types = {
+            migraphx::shape::half_type, migraphx::shape::float_type, migraphx::shape::double_type};
+        if(not contains(supported_score_weight_types, scores_shape.type()))
+        {
+            MIGRAPHX_THROW(
+                "softmaxcrossentropyloss: Score must be either half, float, or double type");
+        }
+
+        if(scores_shape.lens()[0] != label_shape.lens()[0])
+        {
+            MIGRAPHX_THROW(
+                "softmaxcrossentropyloss: Score and Labels must identical batch size inputs");
+        }
+
+        if((scores_shape.ndim() - 1) != label_shape.ndim())
+        {
+            MIGRAPHX_THROW(
+                "softmaxcrossentropyloss: Score and Labels must contain identical K-Dimensions");
+        }
```

Comment:
You can also check for all the extra dimensions have same number of elements or not. 
E.g. 
Scores = [N, C, D1, D2....., Dk] 
Lables = [N, d1, d2,....,dk] 
then `D1 * D2.. * Dk ==d1 * d2 *.. * dk`

Shouldn't really be necessary to check because otherwise onnx model generation itself should throw because it would be invalid onnx. 

But since we are checking all the constraints, might as well check this one additionally.

Reply:
Actually this is a good point. I  was playing with dimensions today and added more batches, or classes and found another issue. Valid or not it's likely worth checking since its assumed based on the spec that D1-Dk are the same, and thus mismatches would be an error

Reply:
Good point. I'll add that check and associated tests

Reply:
Added the case and updated tests accordingly. Added another onnx file too